### PR TITLE
Broadcast field types

### PR DESCRIPTION
**Pull request by otoolep**
Definitely WIP

### CODE REVIEW

**Comment by otoolep** [comment]
@benbjohnson for comment.

This change adds a "broadcast new field type" message to the system. A "field type" is the knowledge, for example, that Meaurement "cpu" has a value "load" and its type is float.

On every write the system checks if a new field has appeared for the point's Measurement. If so, it sends identifies the type and sends out a message so all data node Metastores are updated.

This had the added benefit of reducing our system to a single write code-path for all data, which is a nice simplification. _One write path to rule them all._

I need to add more testing, so am not going to merge this just yet even though the build is green. I also need to take another look at the locks.

File: influxql/ast.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ func InspectDataType(v interface{}) DataType {
 	switch v.(type) {
 	case float64:
 		return Number
+	case int:
```
Comment (otoolep) [inline influxql/ast.go]: I think we want this.

**Comment by pauldix** [comment]
Looks good. There's an optimization we should do where we create measurement, tags, and fields all in one go, but we can save that for later. I'm actually going to do that so that it can create many measurements, tags, and fields in a single command as a write optimization.

Once you have the test for CreateFields, :ship:

**Comment by otoolep** [comment]
Thanks for the review @pauldix -- I did add the unit test, but had to add it at the HTTP handler level, since the new code is unexported. See https://github.com/influxdb/influxdb/commit/02db5a34841298fef1fe96822ebcb5b1bd912ada

**Comment by otoolep** [comment]
Green build, merging now.

**Comment by otoolep** [comment]
https://github.com/influxdb/influxdb/pull/1613 is another related PR.

**Comment by otoolep** [comment]
Unit test code only, self-merging on green build.

**Comment by otoolep** [comment]
Unit test code only, self-merging on green build.

**Pull request merged**